client/src/components/OverviewFolder/CarouselThumbnail.jsx

```diff
@@ -1,9 +1,10 @@
-import React, {useState} from 'react';
+import React, {useState, useEffect} from 'react';
 
 
 const CarouselThumbnail = (props) => {
 
   let [index, setIndex] = useState(0);
+  let [selectedPhoto, setSelectedPhoto] = useState(props.photos[0].thumbnail_url);
```

Comment:
Have you considered passing the  'mainPhoto' state and a function (that changes the mainPhoto) as a prop here  that you call with onClick with each thumbnail